tests/foreman/api/test_environment.py

```diff
@@ -30,6 +30,83 @@
 from robottelo.utils.datafactory import valid_environments_list
 
 
+@pytest.mark.e2e
+@pytest.mark.tier1
+@pytest.mark.upgrade
+def test_positive_CRUD_with_attributes(
+    session_puppet_enabled_sat, module_puppet_org, module_puppet_loc
+):
+    """Check if Environment with attributes can be created, updated and removed
+
+    :id: d2187971-86b2-40c9-a93c-66f37691ae2c
+
+    :BZ: 1337947
+
+    :expectedresults:
+        1. Environment is created and has parameters assigned
+        2. Environment can be listed by parameters
+        3. Environment can be updated
+        4. Environment can be removed
+    """
+    puppet_sat = session_puppet_enabled_sat
+    # Create with attributes
+    env_name = gen_string('alpha')
+    environment = puppet_sat.api.Environment(
+        name=env_name, organization=[module_puppet_org], location=[module_puppet_loc]
+    ).create()
+    assert env_name == environment.name
+    assert len(environment.organization) == 1
+    assert environment.organization[0].id == module_puppet_org.id
+    assert len(environment.location) == 1
+    assert environment.location[0].id == module_puppet_loc.id
+
+    # List by name
+    envs = puppet_sat.api.Environment().search(query=dict(search=f'name={env_name}'))
+    assert len(envs) == 1
+    assert envs[0].name == env_name
+
+    # List by org loc id
+    envs = puppet_sat.api.Environment().search(
+        query=dict(
+            search=f'organization_id={module_puppet_org.id} and location_id={module_puppet_loc.id}'
+        )
+    )
+    assert env_name in [res.name for res in envs]
+
+    # List by org loc name
+    envs = puppet_sat.api.Environment().search(
+        query=dict(
+            search=f'organization={module_puppet_org.name} and location={module_puppet_loc.name}'
+        )
+    )
+    assert env_name in [res.name for res in envs]
+
+    # Update org and loc
+    new_org = puppet_sat.api.Organization().create()
+    new_loc = puppet_sat.api.Location().create()
+    env = puppet_sat.api.Environment(id=environment.id, organization=[new_org]).update(
+        ['organization']
+    )
+    assert len(env.organization) == 1
+    assert env.organization[0].id == new_org.id
+    assert env.organization[0].id != module_puppet_org.id
+
+    env = puppet_sat.api.Environment(id=environment.id, location=[new_loc]).update(['location'])
```

Comment:
why are we making two different calls? one for org and location?

Reply:
we can't directly update both in one call i guess, one way is to read first and then update both, so it will be again two calls, and here we update org first and then loc

Reply:
`puppet_sat.api.Environment(id=environment.id, organization=[new_org], location=[new_loc]).update(['organization', 'location'])`

Reply:
It works, updated, thanks for the suggestion!